Sources/Algorithms/LazySplitCollection.swift

```diff
@@ -0,0 +1,330 @@
+  /// Elements that are used to split the collection are not returned as part of any
+  /// subsequence (except possibly the last one, in the case where `maxSplits` is
+  /// less than the number of separators in the collection).
+  ///
+  /// The following examples show the effects of the `maxSplits` and
+  /// `omittingEmptySubsequences` parameters when lazily splitting a string using a
+  /// closure that matches spaces. The first use of `split` returns each word
+  /// that was originally separated by one or more spaces.
+  ///
+  ///     let line = "BLANCHE:   I don't want realism. I want magic!"
+  ///     for spaceless in line.lazy.split(whereSeparator: { $0 == " " }) {
+  ///       print(spaceless)
+  ///     }
+  ///     // Prints
+  ///     // BLANCHE:
+  ///     // I
+  ///     // don't
+  ///     // want
+  ///     // realism.
+  ///     // I
+  ///     // want
+  ///     // magic!
+  ///
+  /// The second example passes `1` for the `maxSplits` parameter, so the
+  /// original string is split just once, into two new strings.
+  ///
+  ///     for spaceless in line.lazy.split(maxSplits: 1, whereSeparator: { $0 == " " }) {
+  ///       print(spaceless)
+  ///     }
+  ///     // Prints
+  ///     // BLANCHE:
+  ///     // I don't want realism. I want magic!
+  ///
+  /// The final example passes `false` for the `omittingEmptySubsequences`
+  /// parameter, so the returned array contains empty strings where spaces
+  /// were repeated.
+  ///
+  ///     for spaceless in line.lazy.split(omittingEmptySubsequences: false, whereSeparator: { $0 == " " }) {
+  ///       print(spaceless)
+  ///     }
+  ///     // Prints
+  ///     // BLANCHE:
+  ///     //
+  ///     //
+  ///     // I
+  ///     // don't
+  ///     // want
+  ///     // realism.
+  ///     // I
+  ///     // want
+  ///     // magic!
+  ///
+  /// - Parameters:
+  ///   - maxSplits: The maximum number of times to split the collection, or
+  ///     one less than the number of subsequences to return. If
+  ///     `maxSplits + 1` subsequences are returned, the last one is a suffix
+  ///     of the original collection containing the remaining elements.
+  ///     `maxSplits` must be greater than or equal to zero. The default value
+  ///     is `Int.max`.
+  ///   - omittingEmptySubsequences: If `false`, an empty subsequence is
+  ///     returned in the result for each pair of consecutive elements
+  ///     satisfying the `isSeparator` predicate and for each element at the
+  ///     start or end of the collection satisfying the `isSeparator`
+  ///     predicate. The default value is `true`.
+  ///   - whereSeparator: A closure that takes an element as an argument and
+  ///     returns a Boolean value indicating whether the collection should be
+  ///     split at that element.
+  /// - Returns: A lazy sequence of subsequences, split from this collection's
+  ///   elements.
+  ///
+  /// - Complexity: O(*n*), where *n* is the length of the collection.
```

Comment:
Nit:
```suggestion
  /// - Complexity: O(*n*), where *n* is the count of the collection.
```

Reply:
"length" comes from the `split` method comments in Collection.swift: https://github.com/apple/swift/blob/93a28daa3c7d67d0e0a008a00929c32a84b05a5f/stdlib/public/core/Collection.swift#L1528. I have no strong preference in this case, but in general I'm inclined to keep the docs for these lazy splits as close to the existing ones as possible. Is there a doc prose style guide that we could defer to?

Reply:
Ah no worries that is not that important, that was just a suggestion because to me its more reasonable to refer to count for sequences and collections in Swift since the property is `count` ... but if `length` is how it is on other docs we can just keep it :))

Reply:
That is a very compelling argument. I'm less inclined to defer to the existing docs now. 🤔

Reply:
Either phrasing can be argued for, but "length" is used in this context ubiquitously in the rest of the Algorithms package, so let's keep it consistent 🙂

Reply:
IIRC, we use `length` for most of these because sequences don't have a `count`, per se. They're largely interchangeable otherwise!